app/api/odm/MultiTenantMongooseModel.ts

```diff
@@ -1,10 +1,16 @@
-import mongoose, { Schema, UpdateQuery, ModelUpdateOptions } from 'mongoose';
-import { WithId, UwaziFilterQuery } from './models';
+import mongoose, { Schema } from 'mongoose';
+import {
+  DataType,
+  UwaziFilterQuery,
+  UwaziUpdateQuery,
+  UwaziQueryOptions,
+  PartialDataType,
+} from './model';
 import { tenants } from '../tenants/tenantContext';
 import { DB } from './DB';
 
 class MultiTenantMongooseModel<T> {
-  dbs: { [k: string]: mongoose.Model<WithId<T> & mongoose.Document> };
+  dbs: { [k: string]: mongoose.Model<DataType<T>> };
```

Comment:
In the first iteration, `DataType` included an `& Document`.  This is no longer true.  This means that `dbs` no longer explicitly has the `mongoose.Document` type extended, but I think that `T` should already have that passed along on instance.